src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1610,9 +1587,13 @@ namespace cryptonote
   //---------------------------------------------------------------
   bool is_valid_decomposed_amount(uint64_t amount)
   {
-    const uint64_t *begin = valid_decomposed_outputs;
-    const uint64_t *end = valid_decomposed_outputs + sizeof(valid_decomposed_outputs) / sizeof(valid_decomposed_outputs[0]);
-    return std::binary_search(begin, end, amount);
+    if (0 == amount)
+      return false;
+
+    while (amount % 10 == 0)
```

Comment:
Thanks @jeffro256 for this PR. It is great. I have a few comments that we can talk about. 

a. It would be great if we could add a small benchmark, and test included in this PR.
b. While a `while` loop is simple and easy to read. I believe we can test the algorithm performance: 

```
inline // better to have FORCEINLINE here
int c_t_z(uint64_t n) {
    int zeros = 0;
    if (n % 10000000000000000ULL == 0) { zeros += 16; n /= 10000000000000000ULL; }
    if (n % 100000000 == 0) { zeros += 8; n /= 100000000; }
    if (n % 10000 == 0) { zeros += 4; n /= 10000; }
    if (n % 100 == 0) { zeros += 2; n /= 100; }
    if (n % 10 == 0) { zeros += 1; }
    return zeros;
}
amount >>= zeros;
```

I am not sure how something like this would perform in comparison with `while` loop. But worth looking at.

I will ping you on Matrix, to talk about these, even maybe we can work together on this.